woocommerce/google-listings-and-ads
# Add products reporting data source picker and connect all UI query interactions to page route

### Changes proposed in this Pull Request:

Implemented query related UI parts of #244

- Add a new filter config to display the data source picker
- Enable sorting and comparison products functions, and update query to URL for the **CompareProductsTableCard** component
- Check whether has any active paid campaign before showing the products report page and switch available metrics respectively

❗ Please note that:

- This PR doesn't include data connecting, so the results are still mocked data and wouldn't reflect the query state.
- Unused mock data will be removed with another PR that handles the data connecting part.

### Screenshots:

#### 🎥 . Has an active paid campaign

https://user-images.githubusercontent.com/17420811/117132051-def1e980-add4-11eb-93d4-78cdd78eb198.mp4

#### 🖼️ . No active paid campaign

![image](https://user-images.githubusercontent.com/17420811/117132377-5c1d5e80-add5-11eb-93da-ea2c56cd2ffc.png)


### Detailed test instructions:

💡. To test with/without an active paid campaign, you can add a code fragment to [useAdsCampaigns hook](https://github.com/woocommerce/google-listings-and-ads/blob/d960d646026b7d5cbbf1409bfa2db87d9679fede/js/src/hooks/useAdsCampaigns.js#L15) to switch between.

```js
return useSelect( ( select ) => {
	return {
		loaded: true,

		/* Has an active paid campaign */
		data: [ { status: 'enabled' } ],

		/* No active paid campaign */
		// data: [],
	};
```

1. Head to the Products reporting page: `/admin.php?page=wc-admin&path=%2Fgoogle%2Freports%2Fproducts`
2. Change data source by **Show data from** selector and available metrics should be updated respectively
3. All UI query interactions should update to the URL query
    - Switch data source by **Show data from** selector
    - Filter products by **Show** selector
    - Compare products by **Compare** button
    - Change sorting field and order by clicking on the headers of the products table
4. It should keep the same query state on UIs after reloading page
